### PR TITLE
Adding Fake Allied Barracks

### PR DESCRIPTION
**Pull request by PunkPun**
A long requested feature that is largely agreed upon

Cameo was taken from SoScared's "Unplugged" mod

### CODE REVIEW

File: mods/ra/rules/fakes.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FPWR:
+﻿FPWR:
```
Comment (Punsho) [inline mods/ra/rules/fakes.yaml]: I'm not sure what causes this

**Comment by MustaphaTR** [comment]
At the point i feel like we should add fake refinery and helipad too and complete the set.

File: mods/ra/rules/fakes.yaml (hunk 1 above)
Comment (GraionDilach) [reply mods/ra/rules/fakes.yaml]: Looks like an addition/removal of the UTF-8 byte-order mark. If you use Notepad++, make sure the bottom reports UTF-8 and not UTF-8-BOM.

image

Comment (Punsho) [reply mods/ra/rules/fakes.yaml]: I'm using Visual Studio Community and by default it is already encoded Unicode UTF-8

Comment (GraionDilach) [reply mods/ra/rules/fakes.yaml]: VS Community adds BOM by default.

https://stackoverflow.com/questions/5406172/utf-8-without-bom

Comment (Punsho) [reply mods/ra/rules/fakes.yaml]: for now used a different editor

**Comment by netnazgul** [comment]
@MustaphaTR overloading France's defence tab should be considered here. Fake tent is quite a requested feature to deny though.

**Comment by Orb370** [comment]
*rattle rattle*

Long requested feature. If overloading the defense tab is an issue (though we have space for 1 more structure), we could easily remove f. naval yard. Also there's a fake subpen for some reason? (In debug)

**Comment by MustaphaTR** [comment]
Fake Subpen is something original had for some reason too, which is why we have it. It was not buildable there either. I think fakes were meant to be availabe to both sides at some point, sine Shipyard/Subpen was the only side specific fake in the original, others were ConYard, WFac and Radar. If it is an issue i can be `-Buildable:`d instead of `Prerequisite: ~disabled`.

File: mods/ra/rules/fakes.yaml [hunk 2]
```diff
@@ -393,3 +393,39 @@ FACF:
...
 			BottomRight: 1536, 1536
+
+TENF:
+	Inherits: ^FakeBuilding
+	Inherits@infiltrate: ^InfiltratableFake
+	Inherits@shape: ^2x2Shape
+	Selectable:
+		Bounds: 48,48
+	Buildable:
+		BuildPaletteOrder: 871
```
Comment (pchote) [inline mods/ra/rules/fakes.yaml]: Please adjust the `BuildPaletteOrder`s here and in the other fake structures to follow the standard convention - see previous discussion in #16613 for reasoning.

**Review by pchote — COMMENTED** [review]
One request, otherwise LGTM

File: mods/ra/rules/fakes.yaml (hunk 2 above)
Comment (Punsho) [reply mods/ra/rules/fakes.yaml]: Updated. As it seems there was already a place for it

File: mods/ra/rules/fakes.yaml [hunk 3]
```diff
@@ -32,6 +32,42 @@ FPWR:
...
+	Inherits: ^FakeBuilding
+	Inherits@infiltrate: ^InfiltratableFake
+	Inherits@shape: ^2x2Shape
+	Selectable:
+		Bounds: 48,48
+	Buildable:
+		BuildPaletteOrder: 880
+		Queue: Defense
+		Prerequisites: ~structures.france, ~techlevel.infonly
+		Description: Looks like a Tent.
```
Comment (pchote) [inline mods/ra/rules/fakes.yaml]: This is still referencing Tent. Please change to "Looks like an Allied Barracks." to match the other fake descriptions.

Comment (Punsho) [reply mods/ra/rules/fakes.yaml]: 👍

**Review by pchote — APPROVED** [review]

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/db365813d4d40dc203ef0fd73a8f247f77745336...509a1dda83fad68ec8ab839ab4136f2146580ec7

**Comment by Jehoel** [comment]
> @netnazgul  overloading France's defence tab should be considered here.

I'd like to see "groups" or horizontal dividers within each tab - which is handy when you have both Allied and Soviet tech-trees available - to keep Allied and Soviet-exclusive build buttons separate - it could also be used to separate Fake structures from real structures - or move Fake structures to their own separate construction tab?

Or have a single construction tab split into sub-groups? That would also make a better use of the increased screen real-estate on large modern displays.

(With apologies for being off-topic)

**Pull request merged**